functional/FunctionalTest.java

```diff
@@ -3795,6 +3798,155 @@ public static void getDefaultRetention_test() throws Exception {
     }
   }
 
+  /** Test: setObjectRetention(SetObjectRetentionArgs args). */
+  public static void setObjectRetention_test1() throws Exception {
+    String methodName = "setObjectRetention(SetObjectRetentionArgs args)";
+    if (!mintEnv) {
+      System.out.println("Test: " + methodName);
+    }
+
+    long startTime = System.currentTimeMillis();
+    String bucketName = getRandomName();
+    String objectName = getRandomName();
+    try {
+      client.makeBucket(MakeBucketArgs.builder().bucket(bucketName).objectLock(true).build());
+      try {
+        ZonedDateTime retentionUntil = ZonedDateTime.now(Time.UTC).plusDays(1).withNano(0);
+        Retention expectedConfig = new Retention(RetentionMode.GOVERNANCE, retentionUntil);
+
+        try (final InputStream is = new ContentInputStream(1 * KB)) {
+          PutObjectOptions options = new PutObjectOptions(1 * KB, -1);
+          options.setContentType(customContentType);
+          client.putObject(bucketName, objectName, is, options);
+        }
+
+        client.setObjectRetention(
+            SetObjectRetentionArgs.builder()
+                .bucket(bucketName)
+                .object(objectName)
+                .config(expectedConfig)
+                .build());
+        // Set empty object lock configuration by setting bypassGovernanceMode as true
+        Retention emptyConfig = new Retention();
+        client.setObjectRetention(
+            SetObjectRetentionArgs.builder()
+                .bucket(bucketName)
+                .object(objectName)
+                .config(emptyConfig)
+                .bypassGovernanceMode(true)
+                .build());
+
+      } finally {
+        client.removeObject(
+            RemoveObjectArgs.builder().bucket(bucketName).object(objectName).build());
+        client.removeBucket(RemoveBucketArgs.builder().bucket(bucketName).build());
+      }
+      mintSuccessLog(methodName, null, startTime);
+    } catch (Exception e) {
+      handleException(methodName, null, startTime, e);
+    }
+  }
+
+  /** Test: getObjectRetention(GetObjectRetentionArgs args). */
+  public static void getObjectRetention_test1() throws Exception {
+    String methodName = "getObjectRetention(GetObjectRetentionArgs args)";
+    if (!mintEnv) {
+      System.out.println("Test: with shortened retention period: " + methodName);
```

Comment:
This test is also checking to shortening retention until. You could just have `"Test: "`